src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -79,22 +79,26 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetFirmwareRevisionString
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize, size_t & serialNumLen)
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize)
 {
     CHIP_ERROR err;
-    err = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
+    size_t serialNumLen = 0; // including null-terminator
+    err                 = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
+
+    ReturnErrorCodeIf(serialNumLen > bufSize, CHIP_ERROR_BUFFER_TOO_SMALL);
```

Comment:
This is assuming callee does not touch serialNumLen on error conditions.  Should this line move to after the CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND block and to after the check for failing err that should be there?

Also, if serialNumLen is length not including null terminator, then this check should be `>=`, not `>`.  We need to make sure we get this straight.

Reply:
fixed.